ADR/ADR-001-Git-Branching-Model.md

```diff
@@ -0,0 +1,225 @@
+# Architectural Decision Record 001: Git Branching Model
+
+## Status
+[**Proposed** | <s>Accepted</s> | <s>Deprecated</s> | <s>Superseded by [ADR-XXX]</s>]
+
+## Date
+2025-06-25
+
+## Context
+
+We would like to provide guidelines for a more consistent Git branching strategy across repositories to ensure efficient collaboration, reliable releases, and maintainable codebases. We are evaluating two well-established approaches:
+
+Git Flow - The branching model by Vincent Driessen using both main/master and develop branches
+GitHub Flow - GitHub's lightweight branch-based workflow using only main/master
+
+The Git Flow model has been so far the choice for many operational software packages, providing a structured approach to managing releases and features, as well as support branches for hotfixes and long-term maintenance. It has allowed for parallel development of features while maintaining a stable main branch for production releases.
+
+Meanwhile, GitHub Flow has gained popularity for its simplicity and alignment with continuous delivery practices, especially in web applications. It allows for rapid development and deployment cycles, but requires strong discipline to ensure the main branch remains stable and deployable.
+
+We recognise that the choice of branching strategy significantly impacts development velocity, release management, code stability, and team coordination. Different repositories may have varying complexity levels and operational requirements, necessitating a flexible approach that can evolve with project needs.
+
+**Decision Drivers**
+
+* Development team size and experience levels
+* Release frequency and deployment patterns
+* Code review and quality assurance processes
+* Operational complexity and risk tolerance
+* Repository maturity and usage patterns
+* Need for supporting multiple software versions
+
+### Options Considered
+
+#### Option 1: Git Flow (Vincent Driessen's Model)
+
+Based on the branching model described at https://nvie.com/posts/a-successful-git-branching-model/
+
+**Structure**
+
+* main/master: Production-ready code, tagged releases
+* develop: Integration branch for next release containing latest delivered development changes
+* feature branches: Individual feature development, branched from develop
+* release branches: Preparation for production releases, branched from develop
+* hotfix branches: Critical production fixes, branched from main/master
+* support branches: Long-term maintenance for older releases
+
+**Advantages**
+
```

Comment:
I would add here that this workflow model is good for synchronised releases of multiple packages/repositories as you can test all the packages together on develop before doing the synchronised release.
It is also easier to define triggers for longer (integration) tests, as you can use a PR develop->main to trigger the workflow.

Reply:
good point. added.